qgis/examples/elevation_tile_downloader/core/task.py

```diff
@@ -1,117 +1,122 @@
+                directory = self.output_dir.joinpath(dir_name)
+                directory.mkdir(exist_ok=True)
+                tile_name = "%s%02d%s%03d.hgt" % (
+                    hemisphere,
+                    abs(y),
+                    "W" if x < 0 else "E",
+                    abs(x),
+                )
+                filepath = self.output_dir.joinpath(dir_name, tile_name)
+                self.tiles.append(filepath)
+                if filepath.is_file():
+                    QgsMessageLog.logMessage(
+                        message=f"Already downloaded tile {tile_name}",
+                        level=Qgis.Info,
+                    )
+                    continue
+
+                # get the new tile if it didn't exist
+                url = QUrl(
+                    f"http://s3.amazonaws.com/elevation-tiles-prod/skadi/{dir_name}/{tile_name}.gz"
+                )
+                request = QNetworkRequest(url)
+                reply: QgsNetworkReplyContent = nam.blockingGet(request)
+                # turn the reply into a file object and write the decompressed file to disk
+                with gzip.GzipFile(
+                    fileobj=BytesIO(reply.content()), mode="rb"
+                ) as gz:
+                    with open(filepath, "wb") as f:
+                        f.write(gz.read())
+
+                self.setProgress((i / len(self.grid)) * 100)
+
             return True
         except Exception as e:
             self.exception = e
-            self.traceback = traceback.format_exc()
             return False
 
     def finished(self, result):
-        if result:
-            QgsMessageLog.logMessage(
-                message=f'Finished downloading tiles. {len(self.tiles)} tiles downloaded.',
-                level=Qgis.Info
-            )
-            if self.add:
-                self.add_to_map()
-        elif not result and self.exception:
+        """Postprocessing after the run() method."""
+        if self.isCanceled():
+            # if it was canceled by the user
             QgsMessageLog.logMessage(
-                message=f'An exception occurred during downloading: {self.exception}\n'
-                        f'{self.traceback}',
-                level=Qgis.Warning
+                message=f"Canceled download task. {len(self.tiles)} tiles downloaded.",
+                level=Qgis.Warning,
             )
-        elif not result and not self.exception:
-            QgsMessageLog.logMessage(
-                message=f'Canceled download task. {len(self.tiles)} tiles downloaded.',
-                level=Qgis.Warning
+            return
+        elif not result:
+            # if there was an error
+            QMessageBox.critical(
+                iface.mainWindow(),
+                "Tile download error",
+                f"The following error occurred:\n{self.exception.__class__.__name__}: {self.exception}",
             )
+            return
```

Comment:
if an exception occurs it should be more obvious than a log message IMO. lots of people don't even know of that panel's existence I'd bet. this will pop up in the user's face;)

Reply:
good point, hadn't thought of that